### ISSUE TITLE
[Feature] Refactor Python agent UUID generation

### ISSUE THREAD

**Issue by Superskyyy** [apache/skywalking#10849]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Refactoring the uuid generator shows a 30% thoroughput increase in small transactions. UUID in Python agent previously was implemented as a ID class that creates a new UUID1 instance every single time when required, a huge number of uuids is added up as a costly bottleneck that consume large portion of CPU time. New method will replicate the Java agent logic and use a global PROCESSID (uuid4) + ThreadID + Timestamp + (1-9999) as the ids plus other python-specific optimizations.

- [x] - UUID generation change
- [x] - Consider forking behavior that is different from Java applications
- [ ] - Optimize other parts in the tracing core.

### Use case

Reduce performance cost of introducing Python agent.

This probably also leads to a new release along with several smaller enhancements.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
Timstamp + rolling increasing number may face conflict in high concurrency cases. Be careful of that. Java agent includes thread ID to avoid that.

**Comment by wu-sheng** [comment]
Good to see. I am aware of UUID cost in Java. Good to know the same in Python.
@mrproliu I think it is worth to check on Go agent as well.

**Comment by Superskyyy** [comment]
> Good to see. I am aware of UUID cost in Java. Good to know the same in Python. @mrproliu I think it is worth to check on Go agent as well.

This issue applies to most agents, including skywalking-go (I think)

**Comment by mrproliu** [comment]
> > Good to see. I am aware of UUID cost in Java. Good to know the same in Python. @mrproliu I think it is worth to check on Go agent as well.
> 
> This issue applies to most agents, including skywalking-go (I think)

Sure, it looks like a simple version of the snowflake id generator, Am I right? I need to do some performance testing in skywalking-go.

**Comment by wu-sheng** [comment]
Yes, it was inspired by that but without coordinator to assign the range. Instead, we tradeoff ID length for a better performance.

**Comment by Superskyyy** [comment]
Done on Go and Python side, do other agents need an update? @wu-sheng should i keep this issue open as a reminder.

**Comment by wu-sheng** [comment]
It is fine. If people care, they would follow up. Feel free to close once your side is done.

**Comment by Superskyyy** [comment]
> It is fine. If people care, they would follow up. Feel free to close once your side is done.

Okay!

**Comment by wu-sheng** [comment]
This has been
